server/routes/imageUpload.js

```diff
@@ -0,0 +1,28 @@
+const express = require("express");
+const { validate } = require("../middlewares/validate");
+const router = express.Router();
+const upload = require("../services/imageUpload");
+const uploadFile = upload.single("image");
+
+const uploadImage = (req, res) => {
+  uploadFile(req, res, (err) => {
+    if (err) {
+      console.log(err);
+      return res.json({
+        success: false,
+        errors: {
+          title: "Image Upload Error",
+          detail: err.message,
+          error: err,
+        },
+      });
+    } else {
+      const imageUrl = req.file.location;
+      res.status(200).json({ imageUrl: imageUrl });
+    }
+  });
+};
+
+router.post("/", [validate, uploadImage]);
```

Comment:
This is personal preference, but if you only have a couple of functions to run, having the array is not 100% necessary. You can just separate it by commas it would be just fine.

```javascript
router.post("/", validate, uploadImage);
```

To me, having an array of middlewares could be useful if you have a group of middleware functions that you want to run on a specific route, then you could do:

```javascript
const authMiddleware = [
  validate,
  somePreProcessingOnReqObject,
  somePreProcessingOnResObject,
  isUserAuthenticated,
  // ...
];

router.post("/", authMiddleware, (req, res) => {...});
```

Reply:
yup, I agree, I'll just remove the array.